XenModel/Utils/Helpers.cs

```diff
@@ -352,6 +352,24 @@ public static bool ElyOrGreater(Host host)
                 platform_version != null && productVersionCompare(platform_version, "2.1.1") >= 0;
         }
 
+        public static bool HavanaOrGreaterLTSR71(IXenConnection conn)
+        {
+            return conn == null || HavanaOrGreaterLTSR71(Helpers.GetMaster(conn));
+        }
+
+        /// As Havana platform version is same with Ely and Honolulu, so use product version here
+        /// <param name="host">May be null, in which case true is returned.</param>
+        public static bool HavanaOrGreaterLTSR71(Host host)
+        {
+            if (host == null)
+                return true;
+             string product_version = HostProductVersion(host);
+            return
+                product_version != null && 
+                !FalconOrGreater(host) && 
+                productVersionCompare(product_version, "BRANDING_XC_PRODUCT_7_1_2_VERSION") >= 0;

```

Comment:
It should be "[BRANDING_VERSION_7_1_2]" instead of "BRANDING_XC_PRODUCT_7_1_2_VERSION".
Attention:
- Use square brackets around.
- "[BRANDING_VERSION_7_1_2]" is the one to be replaced,  while "BRANDING_XC_PRODUCT_7_1_2_VERSION" is the replacing variable.